usvm-ts/src/main/kotlin/org/usvm/TSExpressions.kt

```diff
@@ -34,6 +36,95 @@ class TSUndefinedValue(ctx: TSContext) : UExpr<TSUndefinedSort>(ctx) {
     }
 }
 
+/**
+ * Utility class for merging expressions with [UBoolSort] sort.
+ *
+ * Mainly created for [not] function used in TSStateForker.
+ */
+class UJoinedBoolExpr(
+    ctx: TSContext,
+    val exprs: List<UBoolExpr>
+) : UBoolExpr(ctx) {
+    override val sort: UBoolSort
+        get() = ctx.boolSort
+
+    private val joinedExprs = ctx.mkAnd(exprs)
+
+    fun not(): UBoolExpr = ctx.mkAnd(exprs.map(ctx::mkNot))
```

Comment:
Are these conditions big? Should we consider storing these negations inside it?

Reply:
No, they are even theoretically small and this function is rarely used. A broader explanation will be inlined in the code above.